src/js/game/shape_definition.js

```diff
@@ -465,6 +465,43 @@ export class ShapeDefinition extends BasicSerializableObject {
         return new ShapeDefinition({ layers: newLayers });
     }
 
+    /**
+     * Stacks the given shape definition on top.
+     */
+    cloneAndUnstack() {
+        const lowerLayers = this.internalCloneLayers();
+
+        if (this.isEntirelyEmpty()) {
+            assert(false, "Can not stack entirely empty definition");
+        }
+
+        // Check if there is only one layer
+        if (this.layers.length === 1) {
+            return [new ShapeDefinition({ layers: lowerLayers })];
+        }
+
+        let upperLayers = [];
+
+        layerCheck:
```

Comment:
This code can be vastly simplified - simply always split it into two shapes, the first one containing `layers[0 ... layers.length - 2]` and the second one containing just `layers[layers.length-1]`.

In the item processor, then emit both shapes, but for each shape check if `definition.isEntirelyEmpty()` and if so, don't emit anything. This will also make it more consistent